### PR TITLE
mpi4py 3.0.3

### PR DESCRIPTION
**Pull request by ax3l**
mpi4py 3.0.3 adds support for Python 3.8 with new Cython files.
https://bitbucket.org/mpi4py/mpi4py/src/3.0.3/CHANGES.rst

I tried to build 3.0.0 with Python 3.8 and this builds as well, therefor I added no conflict.

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-mpi4py/package.py [hunk 1]
```diff
@@ -13,10 +13,11 @@ class PyMpi4py(PythonPackage):
...
     homepage = "https://pypi.python.org/pypi/mpi4py"
-    url      = "https://pypi.io/packages/source/m/mpi4py/mpi4py-3.0.0.tar.gz"
+    url      = "https://pypi.io/packages/source/m/mpi4py/mpi4py-3.0.3.tar.gz"
     git      = "https://github.com/mpi4py/mpi4py.git"
 
     version('develop', branch='master')
+    version('3.0.3', sha256='012d716c8b9ed1e513fcc4b18e5af16a8791f51e6d1716baccf988ad355c5a1f')
     version('3.0.1', sha256='6549a5b81931303baf6600fa2e3bc04d8bd1d5c82f3c21379d0d64a9abcca851')
     version('3.0.0', sha256='b457b02d85bdd9a4775a097fac5234a20397b43e073f14d9e29b6cd78c68efd7')
     version('2.0.0', sha256='6543a05851a7aa1e6d165e673d422ba24e45c41e4221f0993fe1e5924a00cb81')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-mpi4py/package.py]: Couple minor updates:

- The cython version required is `0.22:`
- I think the python version requirement should be `2.6:2.8,3.2:`

Comment (ax3l) [reply var/spack/repos/builtin/packages/py-mpi4py/package.py]: Updated, although Python 2.6 as well as 3.2-3.4 are already EOL:
https://devguide.python.org/devcycle/#end-of-life-branches

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-mpi4py/package.py]: Python 2.7 will be EOL soon but we'll probably end up supporting it for quite a while...

Comment (ax3l) [reply var/spack/repos/builtin/packages/py-mpi4py/package.py]: `*adds mixed feeling emoji*` ;-)

**Pull request merged**